pySNOM/images.py

```diff
@@ -142,29 +145,67 @@ def transform(self, data):
         else:
             return data - norm
 
+class RotatePhase(Transformation):
 
-class Process:
-    def __init__(self, data = None):
-        self.data = data
-        if self.data is not None:
-            self.output = np.zeros(np.shape(self.data))
+    def __init__(self, degree=90.0):
+        self.degree = degree
+
+    def transform(self, amplitudedata, phasedata):
+        # Construct complex dataset
+        complexdata = amplitudedata * np.exp(phasedata*complex(1j))
+        # Rotate and extract phase
+        return np.angle(complexdata*np.exp(np.deg2rad(self.degree)*complex(1j)))
+
+class SelfReference(Transformation):
+
+    def __init__(self, datatype=DataTypes.Phase):
+        self.datatype = datatype
+
+    def transform(self, data, referencedata):
+        if self.datatype == DataTypes.Amplitude:
+            return np.divide(data, referencedata)
+        elif self.datatype == DataTypes.Phase:
+            return data-referencedata
         else:
-            self.output = None
+            # TODO: We should replace this with a datatype error raise
+            print("Self-referencing makes only sense for amplitude or phase data")
 
-    def setData(self, data):
-        self.data = data
-        self.output = np.zeros(np.shape(self.data))
+class SimpleNormalize(Transformation):
 
-    def getOutput(self):
-        return self.output
-    
-    def getData(self):
-        return self.data
+    def __init__(self, method='median', value=1, datatype=DataTypes.Phase):
+        self.method = method
+        self.value = value
+        self.datatype = datatype
 
-    def bg_polyfit(self, xorder = int(1), yorder = int(1), datatype = DataTypes.Phase):
-        Z = copy.deepcopy(self.data)
-        x = list(range(0, self.data.shape[0]))
-        y = list(range(0, self.data.shape[1]))
+    def transform(self, data):
```

Comment:
Ok, I do not know what this one is supposed to do, but be aware that not the complete data matrix is divided by its average. If you have multiple columns/rows this is unlikely to be what you want.

Reply:
I was actually concerned about the code that follows the comment, namely `data / np.median(data)` and similar.